lib/tools/apk-utils.js

```diff
@@ -142,6 +147,13 @@ apkUtilsMethods.startApp = async function startApp (startAppOptions = {}) {
       }
       throw new Error(`Activity name '${startAppOptions.activity}' used to start the app doesn't ` +
                       `exist or cannot be launched! Make sure it exists and is a launchable activity`);
+    } else if (stdout.includes('Error: Intent does not match any activities') || stdout.includes('Error: Activity not started, unable to resolve Intent')) {
+      let intentName = startAppOptions.action;
```

Comment:
```
const intentName = `${startAppOptions.action}${startAppOptions.optionalIntentArguments ? ' ' + startAppOptions.optionalIntentArguments : ''}`
```

Reply:
I did that first but it felt harder to read - would you prefer I switched it?

Reply:
yeah, it's quite long. Although, there is probably a need to do the same operation for the name in the exception handler below. Perhaps, it makes sense to extract the formatting into a separate closure?